### PR TITLE
use logservice in bvt

### PR DESCRIPTION
**Pull request by jiangxinmeng1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [x] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/vm/engine/tae/options/types.go [hunk 1]
```diff
@@ -55,8 +64,9 @@ type Options struct {
 	LogtailCfg    *LogtailCfg
 	Catalog       *catalog.Catalog
 
-	Clock clock.Clock
-	Fs    fileservice.FileService
-	Lc    logservice.Client
-	Shard metadata.DNShard
+	Clock     clock.Clock
+	Fs        fileservice.FileService
+	Lc        logservice.Client
```
Comment (zhangxu19830126) [inline pkg/vm/engine/tae/options/types.go:69]: This should be a logservice create factory, and TAE will hold many log service instance to provide high performance.

File: pkg/vm/engine/tae/logstore/store/store.go [hunk 2]
```diff
@@ -45,6 +47,12 @@ type StoreImpl struct {
 	truncateQueue   sm.Queue
 }
 
+func NewStoreWithLogserviceDriver(logserviceCfg *logservice.ClientConfig) Store {
```
Comment (zhangxu19830126) [inline pkg/vm/engine/tae/logstore/store/store.go:50]: Logserivce should rename to LogService.

File: pkg/cnservice/tae.go [hunk 3]
```diff
@@ -40,7 +45,29 @@ func initTAE(
...
+		cancel()
+		if err != nil {
+			panic(err)
+		}
+		opts.Lc = lc
+		opts.LogStoreT = options.LogstoreLogservice
+	case options.LogstoreBatchStore, "":
+		opts.LogStoreT = options.LogstoreBatchStore
+	default:
+		panic(fmt.Sprintf("invalid logstore type: %v", cfg.Engine.Logstore))
```
Comment (fengttt) [inline pkg/cnservice/tae.go:67]: panic a moerr?

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

File: pkg/cnservice/tae.go [hunk 4]
```diff
@@ -40,7 +45,29 @@ func initTAE(
...
+	}
+
+	tae, err := db.Open(targetDir+"/tae", opts)
 	if err != nil {
 		logutil.Infof("Open tae failed. error:%v", err)
```
Comment (fengttt) [inline pkg/cnservice/tae.go:72]: Do you want to return err or panic?   Either way, but better be consistent.

**Review by fengttt — APPROVED** [review]
Minor.   Thank you.

**Review by daviszhen — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

File: pkg/cnservice/tae.go [hunk 5]
```diff
@@ -40,7 +45,29 @@ func initTAE(
...
 
-	tae, err := db.Open(targetDir+"/tae", nil)
+	opts := &options.Options{}
+	switch cfg.Engine.Logstore {
+	case options.LogstoreLogservice:
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		lc, err := logservice.NewClient(ctx, logservice.ClientConfig{
+			ReadOnly:         false,
+			LogShardID:       1,
+			DNReplicaID:      1,
```
Comment (volgariver6) [inline pkg/cnservice/tae.go:55]: Why LogShardID and DNReplicaID are 1?

**Review by XuPeng-SH — APPROVED** [review]
lgtm

File: pkg/vm/engine/tae/options/types.go (hunk 1 above)
Comment (jiangxinmeng1) [reply pkg/vm/engine/tae/options/types.go]: Ok, I'll use a create factory.

File: pkg/vm/engine/tae/logstore/store/store.go (hunk 2 above)
Comment (jiangxinmeng1) [reply pkg/vm/engine/tae/logstore/store/store.go]: OK

File: pkg/cnservice/tae.go (hunk 4 above)
Comment (jiangxinmeng1) [reply pkg/cnservice/tae.go:72]: OK, I'll return moerr.

File: pkg/cnservice/tae.go (hunk 5 above)
Comment (jiangxinmeng1) [reply pkg/cnservice/tae.go]: I'll use config instead.

**Review by zhangxu19830126 — APPROVED** [review]

File: etc/launch-tae-logservice/cn1.toml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+backend = "DISK-ETL"
+data-dir = "store"
+
+[cn]
+uuid = "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf"
+listen-address = "127.0.0.1:3307"
+role = "TP"
+
+[cn.Engine]
+type = "tae"
```
Comment (reusee) [inline etc/launch-tae-logservice/cn1.toml:36]: "tae" means the moengine, this engine may not work with the memory DN.
dn1.toml is not necessary if you want to run moengine only.

File: etc/launch-tae-logservice/dn1.toml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+data-dir = "store"
+
+[dn]
+uuid = "dd4dccb4-4d3c-41f8-b482-5251dc7a41bf"
+listen-address = "127.0.0.1:41010"
+service-address = "127.0.0.1:41010"
+
+[dn.Txn.Storage]
+# txn storage backend implementation. [TAE|MEM]
+backend = "MEM"
```
Comment (reusee) [inline etc/launch-tae-logservice/dn1.toml:37]: "MEM" DN is compatible with "memory" CN or "disttae" CN.
So if you want to run a "tae" CN, this file is not necessary.

File: etc/launch-tae-logservice/launch.toml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+logservices = [
+    "./etc/launch-tae-logservice/log1.toml",
+    "./etc/launch-tae-logservice/log2.toml",
+    "./etc/launch-tae-logservice/log3.toml"
+]
+
+dnservices = [
```
Comment (reusee) [inline etc/launch-tae-logservice/launch.toml:7]: The "tae" CN does not need a DN, this section could be deleted

Comment (jiangxinmeng1) [reply etc/launch-tae-logservice/launch.toml]: Ok, I'll remove this section and dn1.toml

File: etc/launch-moengine-CN-no-DN/dn1.toml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    "127.0.0.1:32021",
+]
+
+[[fileservice]]
+name = "LOCAL"
+backend = "MEM"
+
+[[fileservice]]
+name = "S3"
+backend = "MEM"
```
Comment (nnsgmsone) [inline etc/launch-moengine-CN-no-DN/dn1.toml:23]: why not tae? just mem?

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**